### PR TITLE
Clean up build stuff

### PR DESCRIPTION
**Pull request by cpuguy83**
1. Uses modern go in Dockerfile
2. Fix `make build` pre-reqs
3. Remove interim BUILDFLAGS handling
4. Remove vendor dir (which was unused)
5. passthrough buildkit platform vars
6. parse buildkit vars in makefile

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,8 +1,10 @@
...
 COPY . /go/src/github.com/cpuguy83/go-md2man
 WORKDIR /go/src/github.com/cpuguy83/go-md2man
-RUN CGO_ENABLED=0 go build
+RUN make build
 
 FROM scratch
-COPY --from=build /go/src/github.com/cpuguy83/go-md2man/go-md2man /go-md2man
-ENTRYPOINT ["/go-md2man"]
+COPY --from=build /go/src/github.com/cpuguy83/go-md2man/bin/go-md2man /go-md2man
+ENTRYPOINT ["/go-md2man"]
```
Comment (thaJeztah) [inline Dockerfile:10]: nit: missing newline at end of file

File: Dockerfile [hunk 2]
```diff
@@ -1,8 +1,10 @@
-FROM golang:1.8 AS build
+ARG GO_VERSION=1.17
```
Comment (thaJeztah) [inline Dockerfile:1]: Perhaps we should adjust the versions we test in CI accordingly

(I see the `go.mod` still lists `go 1.11`. that's _probably_ fine, but we don't test that version anymore, so it _could_ be broken without knowing)

File: Dockerfile [hunk 3]
```diff
@@ -1,8 +1,10 @@
...
+
+FROM golang:${GO_VERSION} AS build
 COPY . /go/src/github.com/cpuguy83/go-md2man
 WORKDIR /go/src/github.com/cpuguy83/go-md2man
-RUN CGO_ENABLED=0 go build
+RUN make build
```
Comment (thaJeztah) [inline Dockerfile:6]: Wondering; was there a reason to continue using the makefile for this? If there's no more uses for `BUILD_FLAGS`, I'm wondering if a plain `go build` would work (if needed with  a `-o` to store the binary somewhere else). Looks like building is almost instant, so we could even run `go test` as part of the build, and run a `md2man --version` to verify the binary is not defunct

(edit: erm, except it doesn't have a `--version` or `--help` 🤔 )

File: Makefile [hunk 4]
```diff
@@ -14,12 +14,9 @@ clean:
 test:
 	@go test $(TEST_FLAGS) ./...
 
-bin/go-md2man: actual_build_flags := $(BUILD_FLAGS) -o bin/go-md2man
-bin/go-md2man: bin
-	@CGO_ENABLED=0 go build $(actual_build_flags)
-
-bin:
-	@mkdir ./bin
+bin/go-md2man: go.mod go.sum md2man/* *.go
```
Comment (thaJeztah) [inline Makefile:17]: Can't comment on the lines above, but if we're cleaning up;

- `LINTER_BIN` looks unused
- perhaps change `GO111MODULE` default to `auto`

File: Dockerfile (hunk 3 above)
Comment (cpuguy83) [reply Dockerfile:6]: No reason other than to make sure its the same command.

File: Makefile (hunk 4 above)
Comment (cpuguy83) [reply Makefile:16]: No this needs go modules. We need to make sure its own regardless of if we are in gopath or not.

**Comment by cpuguy83** [comment]
Added go1.11 to the test matrix.
Intentionally keeping older go because this is picked up in distro packages.

File: Makefile (hunk 4 above)
Comment (thaJeztah) [reply Makefile:16]: Hm... looks like it's ignoring the `vendor` directory.

```bash
rm -rf bin && docker run --rm -e GO111MODULE=auto -e CGO_ENABLED=0 -v $(pwd):/app -w /app golang:1.17 sh -c 'go build -o bin/go-md2man && bin/go-md2man --foobar'
go: downloading github.com/russross/blackfriday/v2 v2.1.0
flag provided but not defined: -foobar

rm -rf bin && docker run --rm -e GO111MODULE=on -e CGO_ENABLED=0 -v $(pwd):/app -w /app golang:1.17 sh -c 'go build -o bin/go-md2man && bin/go-md2man --foobar'
go: downloading github.com/russross/blackfriday/v2 v2.1.0
flag provided but not defined: -foobar

rm -rf bin && docker run --rm -e GO111MODULE=auto -e CGO_ENABLED=0 -v $(pwd):/go/src/github.com/cpuguy83/go-md2man -w /go/src/github.com/cpuguy83/go-md2man golang:1.17 bash -c 'go build -o bin/go-md2man && bin/go-md2man --foobar'
go: downloading github.com/russross/blackfriday/v2 v2.1.0
flag provided but not defined: -foobar

rm -rf bin && docker run --rm -e GO111MODULE=on -e CGO_ENABLED=0 -v $(pwd):/go/src/github.com/cpuguy83/go-md2man -w /go/src/github.com/cpuguy83/go-md2man golang:1.17 bash -c 'go build -o bin/go-md2man && bin/go-md2man --foobar'
go: downloading github.com/russross/blackfriday/v2 v2.1.0
flag provided but not defined: -foobar
```

So looks like all permutations ignore `vendor` and will download the module. Disabling networking shows that's indeed the case;

```bash
rm -rf bin && docker run --rm -e GO111MODULE=on --network=none -e CGO_ENABLED=0 -v $(pwd):/go/src/github.com/cpuguy83/go-md2man -w /go/src/github.com/cpuguy83/go-md2man golang:1.17 bash -c 'go build -o bin/go-md2man && bin/go-md2man --foobar'
go: github.com/russross/blackfriday/v2@v2.1.0: Get "https://proxy.golang.org/github.com/russross/blackfriday/v2/@v/v2.1.0.mod": dial tcp: lookup proxy.golang.org on 192.168.65.5:53: dial udp 192.168.65.5:53: connect: network is unreachable
go: downloading github.com/russross/blackfriday/v2 v2.1.0
go: github.com/russross/blackfriday/v2@v2.1.0: Get "https://proxy.golang.org/github.com/russross/blackfriday/v2/@v/v2.1.0.mod": dial tcp: lookup proxy.golang.org on 192.168.65.5:53: dial udp 192.168.65.5:53: connect: network is unreachable
```

The only option that works is to use `-mod=vendor`:

```bash
rm -rf bin && docker run --rm -e GO111MODULE=on --network=none -e CGO_ENABLED=0 -v $(pwd):/go/src/github.com/cpuguy83/go-md2man -w /go/src/github.com/cpuguy83/go-md2man golang:1.17 bash -c 'go build -mod=vendor -o bin/go-md2man && bin/go-md2man --foobar'

flag provided but not defined: -foobar
Usage of bin/go-md2man:
```

Comment (cpuguy83) [reply Makefile:16]: Right yes, vendor is leftover from go mod transition...

Comment (cpuguy83) [reply Makefile:43]: Removed vendor dir.

**Pull request merged**